test/unix/test_sleep_and_timeout.ml

```diff
@@ -4,9 +4,11 @@
 open Test
 open Lwt.Infix
 
+(* None of the APIs make promises about how much larger the elapsed time will
+ * be, but they all promise that it won't be less than the expected time. *)
 let cmp_elapsed_time start_time expected_time =
   let elapsed_time = Unix.gettimeofday () -. start_time in
-  elapsed_time -. expected_time < 0.005
+  elapsed_time -. expected_time >= 0.0
```

Comment:
Ok this is really a nit, but you could write `elapsed_time > expected_time` if keeping this condition :)

I still suggest maybe checking that it's within a factor of two, or maybe 10%, though.